static/shared/js/emoji.js

```diff
@@ -1,5 +1,10 @@
 import _ from "lodash";
 
+/* eslint-disable-next-line import/no-restricted-paths */
+import * as blueslip from "../../js/blueslip.ts";
+/* eslint-disable-next-line import/no-restricted-paths */
+import {page_params} from "../../js/page_params.ts";
```

Comment:
These import restrictions are there for a reason.  

We should remove the `page_params` import by handling the `text` emojiset in the caller, I think. This new function is cleaner returning all the details we have about the emoji, not a modified set depending on the currently configuration.

For blueslip, I think we can potentially change that to a `throw new Error`;